test/test_payu.py

```diff
@@ -301,3 +301,116 @@ def test_list_archive_dirs():
     # Clean up test archive
     shutil.rmtree(tmp_archive)
     shutil.rmtree(tmp_archive_2)
+
+
+@pytest.fixture
+def script_control_dir():
+    # Create a temporary control directory
+    control_dir = tmpdir / 'script_control_dir'
+    control_dir.mkdir()
+
+    yield control_dir
+
+    # Tear down
+    shutil.rmtree(control_dir)
```

Comment:
Does [pytest's tmp-path fixture](https://docs.pytest.org/en/latest/reference/reference.html#tmp-path) provide this functionality?

Reply:
Yes it does! I've updated to use the tmp_path fixture